services/movie_session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+        show_time=movie_show_time
+    )
+
+
+def get_movies_sessions(
+        session_date: datetime = None,
+) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(
+            show_time__date=session_date
+        )
+
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(
+        movie_session_id: int,
+) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(id=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)

```

Comment:
let's use ids directly(re-read checklist again). you can set show_time_id and movie_id fields with ids you get so that you don't need to make additional queries to database